### PR TITLE
Add a generic API

### PR DESCRIPTION
**Pull request by amontoison**
I will add a dispatch for the wrappers `Symmetric` and `Hermitian` tomorrow.

### CODE REVIEW

File: src/generic.jl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  n = LinearAlgebra.checksquare(A)
+  solver = CudssSolver(A, 'G', 'F')
+  x = CudssMatrix(T, n)
+  b = CudssMatrix(T, n)
+  cudss("analysis", solver, x, b)
+  cudss("factorization", solver, x, b)
+  return solver
+end
+
+function LinearAlgebra.ldlt(A::CuSparseMatrixCSR{T,Cint}) where T <: BlasFloat
```
Comment (frapac) [inline src/generic.jl:11]: I am not sure if we should use `ldlt` here, if cudss is using the Bunch-Kaufman factorization for symmetric indefinite matrix. I would suggest renaming this function to `LinearAlgebra.bunchkaufman`.

File: src/generic.jl [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  return solver
+end
+
+for fun in (:lu!, :ldlt!, :cholesky!)
+  @eval begin
+    function LinearAlgebra.$fun(solver::CudssSolver, A::CuSparseMatrixCSR{T,Cint}) where T <: BlasFloat
+      n = LinearAlgebra.checksquare(A)
+      cudss_set(solver.matrix, A)
+      x = CudssMatrix(T, n)
+      b = CudssMatrix(T, n)
```
Comment (frapac) [inline src/generic.jl:39]: Should we allocate two new vectors each time we are updating the factorization?

Comment (amontoison) [reply src/generic.jl:39]: Nop, I just create a dummy `CudssMatrix` for `x` and `b` with these new constructors [here](https://github.com/exanauts/CUDSS.jl/pull/7/files#diff-754e6441f2054c00a80b0953b1876f78a4f3e85993eebd6fa0e3b883611a58caR33-R51).
I use a similar trick with dummy descriptors for `KrylovOperator` and `TriangularOperator` in `KrylovPreconditioners.jl.

File: src/generic.jl (hunk 1 above)
Comment (amontoison) [reply src/generic.jl:11]: NVIDIA answered me this evening and they say that it's an LDL' factorization but I don't understand why we can solve a linear system with [0 1; 1 0] in that case. I will have more news tomorrow.

Comment (frapac) [reply src/generic.jl]: ok, good to know

**Pull request merged**